blaze-server/src/main/scala/org/http4s/server/blaze/Http1ServerStage.scala

```diff
@@ -189,20 +192,24 @@ private[blaze] class Http1ServerStage[F[_]](
       case Right(req) =>
         executionContext.execute(new Runnable {
           def run(): Unit = {
-            val action = Sync[F]
-              .suspend(raceTimeout(req))
+            val action = F
+              .defer(raceTimeout(req))
```

Comment:
I can't recall why we're deferring this.  `raceTimeout` returns in `F`, no?

Reply:
```scala
private[this] val raceTimeout: Request[F] => F[Response[F]] =
    responseHeaderTimeout match {
      case finite: FiniteDuration =>
        val timeoutResponse = F.sleep(finite).as(Response.timeout[F])
        req => F.race(runApp(req), timeoutResponse).map(_.merge)
      case _ =>
        runApp
    }
```
I don't see any strictly evaluated side effects here... so it looks safe to remove